src/assets/AssetsDetectionController.ts

```diff
@@ -84,18 +83,59 @@ export class AssetsDetectionController extends BaseController<AssetsDetectionCon
    */
   name = 'AssetsDetectionController';
 
-  /**
-   * List of required sibling controllers this controller needs to function
-   */
-  requiredControllers = ['AssetsContractController', 'AssetsController', 'NetworkController', 'PreferencesController'];
+  private getOpenSeaApiKey: () => string | undefined;
+
+  private getBalancesInSingleCall: AssetsContractController['getBalancesInSingleCall'];
+
+  private addTokens: AssetsController['addTokens'];
+
+  private addCollectible: AssetsController['addCollectible'];
+
+  private removeCollectible: AssetsController['removeCollectible'];
+
+  private getAssetsState: () => AssetsState;
 
   /**
    * Creates a AssetsDetectionController instance
    *
+   * @param options
+   * @param options.onAssetsStateChange - Allows subscribing to assets controller state changes
+   * @param options.onPreferencesStateChange - Allows subscribing to preferences controller state changes
+   * @param options.onNetworkStateChange - Allows subscribing to network controller state changes
+   * @param options.getOpenSeaApiKey - Gets the OpenSea API key, if one is set
+   * @param options.getBalancesInSingleCall - Gets the balances of a list of tokens for the given address
+   * @param options.addTokens - Add a list of tokens
+   * @param options.addCollectible - Add a collectible
+   * @param options.removeCollectible - Remove a collectible
+   * @param options.initialAssetsState - The initial state of the Assets controller
```

Comment:
This should be updated - `initialAssetsState` was replaced with `getAssetsState`

Reply:
Resolved in c991da5f5e24341742af5a36efa57053459e410a